src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_RankSumTest.java

```diff
@@ -293,18 +295,21 @@ protected void combineAttributeMap(final ReducibleAnnotationData<Histogram> toAd
     }
 
     private String makeReducedAnnotationString(final VariantContext vc, final Map<Allele,Double> perAltRankSumResults) {
-        String annotationString = "";
+        final StringBuilder annotationString = new StringBuilder();
         for (final Allele a : vc.getAlternateAlleles()) {
-            if (!annotationString.isEmpty()) {
-                annotationString += AnnotationUtils.ALLELE_SPECIFIC_REDUCED_DELIM;
+            if (!(annotationString.length() == 0)) {
+                annotationString.append(AnnotationUtils.ALLELE_SPECIFIC_REDUCED_DELIM);
             }
             if (!perAltRankSumResults.containsKey(a)) {
-                logger.warn("ERROR: VC allele not found in annotation alleles -- maybe there was trimming?");
+                logger.warn("VC allele not found in annotation alleles -- maybe there was trimming?  Allele " + a.getDisplayString() + " will be marked as missing.");
+                annotationString.append(VCFConstants.MISSING_VALUE_v4); //add the missing value or the array size and indexes won't check out
```

Comment:
I assume that this was the actual fix? Would it be possible to add a direct unit test for `makeReducedAnnotationString()` if you exposed it as package-accessible?